checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -260,13 +264,17 @@ private boolean canBeSatisfiedThrough(LocalVariableNode localVariableNode) {
           TypeElement typeElt = TypesUtils.getTypeElement(reference.getType());
           if (typeElt == null) {
             // typeElt is null if reference.getType() was not a class, interface, annotation type,
-            // or
-            // enum---that is, was not an annotatable type.
+            // or enum -- that is, was not an annotatable type.
+            // TODO: Why use TOP?  I think it would be better to throw an exception, if this is
+            // something that should not happen.
             // That shouldn't happen, but if it does fall back to a safe default (i.e. top).
             mcAnno = mustCallAnnotatedTypeFactory.TOP;
           } else {
+            // TODO: Do we have an answer for the below question?
```

Comment:
I'll look into it more deeply, but in general there is no guarantee that every Obligation corresponds to a real resource. This code (and the code above that sets `mcAnno` to top when `typeElt` is null) can be viewed as ways to avoid false positives while staying conservative and preserving the guarantees of the analysis.